WordPress/Classes/Utility/Blogging Prompts/ReminderScheduleCoordinator.swift

```diff
@@ -147,12 +147,16 @@ private extension ReminderScheduleCoordinator {
     }
 
     func reminderType(for blog: Blog) -> ReminderType {
-        if Feature.enabled(.bloggingPrompts),
-           let settings = promptReminderSettings(for: blog),
-           settings.promptRemindersEnabled {
-            return .bloggingPrompts
+        guard Feature.enabled(.bloggingPrompts),
+              let settings = promptReminderSettings(for: blog),
+              let context = settings.managedObjectContext else {
+            return .bloggingReminders
         }
 
-        return .bloggingReminders
+        var reminderType: ReminderType = .bloggingReminders
+        context.performAndWait {
+            reminderType = settings.promptRemindersEnabled ? .bloggingPrompts : .bloggingReminders
+        }
+        return reminderType
```

Comment:
If I read it correctly, this method depends only on the state of the feature flags and the input `Blog`. This should make it "easy enough" to unit test, which would greatly increase our confidence in its behavior.

Tip: `Feature.enabled(.bloggingPrompts)` is an hidden dependency on a shared value. We can make it explicit like so:

```swift
func reminderType(for blog: Blog, bloggingPromptsEnabled: Bool = Feature.enabled(.bloggingPrompts)) -> ReminderType {
```

This way, the tests can explicitly set it (e.g. `reminderType(for blog: Blog, bloggingPromptsEnabled: false)`) while the app code can use the `Feature` value like before.

Reply:
Thanks for the suggestion @mokagio! The only issue here is that `reminderType()` is private. I'm not 100% comfortable with making it public just for testing. If you take a look at `ReminderScheduleCoordinatorTests`, you'll see that it's being tested implicitly via `test_scheduled_withPromptsDisabled_returnsScheduleForBloggingReminders` and `test_scheduled_withPromptsEnabled_returnsScheduleForBloggingPrompts` amongst other tests. The tests, however, don't test all the possible scenarios for `reminderType()`. What do you think?